python/TestHarness/RunParallel.py

```diff
@@ -179,22 +180,40 @@ def returnToTestHarness(self, job_index):
         if p.poll() == None: # process has not completed, it timed out
             output += '\n' + "#"*80 + '\nProcess terminated by test harness. Max time exceeded (' + str(tester.specs['max_time']) + ' seconds)\n' + "#"*80 + '\n'
             f.close()
+            tester.setStatus('TIMEOUT', tester.bucket_fail)
             if platform.system() == "Windows":
                 p.terminate()
             else:
                 pgid = os.getpgid(p.pid)
                 os.killpg(pgid, SIGTERM)
 
-            if not self.harness.testOutputAndFinish(tester, RunParallel.TIMEOUT, output, time, clock()):
+            if self.options.pbs and not self.options.processingPBS:
+                if not self.harness.handleTestStatus(tester, output):
+                    did_pass = False
+            elif not self.harness.testOutputAndFinish(tester, RunParallel.TIMEOUT, output, time, clock()):
                 did_pass = False
         else:
             f.close()
 
-            if tester in self.reported_jobs:
-                tester.specs.addParam('caveats', ['FINISHED'], "")
+            # PBS jobs
+            if self.options.pbs and not self.options.processingPBS:
+                if not self.harness.handleTestStatus(tester, output):
+                    did_pass = False
 
-            if not self.harness.testOutputAndFinish(tester, p.returncode, output, time, clock()):
-                did_pass = False
+            # All other jobs
+            else:
+                if tester in self.reported_jobs:
+                    tester.specs.addParam('caveats', ['FINISHED'], "")
+
+                if self.options.dry_run:
+                    # Set the successful message for DRY_RUN
+                    tester.success_message = 'DRY RUN'
+                    output += '\n'.join(tester.processResultsCommand(tester.specs['moose_dir'], self.options))
+                else:
+                    output = tester.processResults(tester.specs['moose_dir'], p.returncode, self.options, output)
+
+                if not self.harness.testOutputAndFinish(tester, p.returncode, output, time, clock()):
+                    did_pass = False
```

Comment:
This section, I would say is the most significant change since the last revert... Just FYI to the reviewer...